src/Nethermind/Nethermind.Merge.AuRa/AuRaPostMergeBlockProducer.cs

```diff
@@ -0,0 +1,53 @@
+using Nethermind.Blockchain;
+using Nethermind.Consensus;
+using Nethermind.Consensus.Processing;
+using Nethermind.Consensus.Producers;
+using Nethermind.Consensus.Transactions;
+using Nethermind.Core;
+using Nethermind.Core.Specs;
+using Nethermind.Logging;
+using Nethermind.Merge.Plugin.BlockProduction;
+using Nethermind.State;
+
+namespace Nethermind.Merge.AuRa
+{
+    public class AuRaPostMergeBlockProducer : PostMergeBlockProducer
+    {
+        public AuRaPostMergeBlockProducer(
+            ITxSource txSource,
+            IBlockchainProcessor processor,
+            IBlockTree blockTree,
+            IBlockProductionTrigger blockProductionTrigger,
+            IStateProvider stateProvider,
+            IGasLimitCalculator gasLimitCalculator,
+            ISealEngine sealEngine,
+            ITimestamper timestamper,
+            ISpecProvider specProvider,
+            ILogManager logManager)
+            : base(
+                txSource,
+                processor,
+                blockTree,
+                blockProductionTrigger,
+                stateProvider,
+                gasLimitCalculator,
+                sealEngine,
+                timestamper,
+                specProvider,
+                logManager)
+        {
+        }
+
+        public override Block PrepareEmptyBlock(BlockHeader parent, PayloadAttributes? payloadAttributes = null)
+        {
+            var block = base.PrepareEmptyBlock(parent, payloadAttributes);
+
+            // processing is only done to apply AuRa block rewards and rewrite posdao contracts
+            // TODO should we return null instead of empty block if we couldn't process?
+            if (TrySetState(parent.StateRoot))
+                block = ProcessPreparedBlock(block, null) ?? block;
+
+            return block;
```

Comment:
If processing fails we still return unprocessed block

Reply:
You think it's better to return null? Maybe throw? I don't know how would it impact the GetPayloadHandler if we return null here, I'll check

Reply:
Let's throw if this happens. It would be a bug.